hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MetricsRegionServer.java

```diff
@@ -122,89 +102,75 @@ public MetricsRegionServerWrapper getRegionServerWrapper() {
     return regionServerWrapper;
   }
 
-  public void updatePutBatch(TableName tn, long t) {
-    if (tableMetrics != null && tn != null) {
-      tableMetrics.updatePutBatch(tn, t);
+  public void ifTableRequestsMetricsExist(HRegion region, Consumer<MetricsTableRequests> consumer) {
+    if (region.getMetricsTableRequests() != null) {
+      consumer.accept(region.getMetricsTableRequests());
     }
+  }
+
+  public void updatePutBatch(HRegion region, long t) {
+    ifTableRequestsMetricsExist(region, metrics -> metrics.updatePutBatch(t));
```

Comment:
can you replace these with normal null checks? similarly not sure we need the overhead of a lambda.

(i had originally suggested a method like this back when the RSRpcServices diff was really large, but now that it's all nicely encapsulated I don't think it's needed.